substrate/primitives/api/proc-macro/src/runtime_metadata.rs

```diff
@@ -173,9 +174,13 @@ pub fn generate_decl_runtime_metadata<'a>(
 		#crate_::frame_metadata_enabled! {
 			#( #attrs )*
 			#[inline(always)]
-			pub fn runtime_metadata #impl_generics (impl_version: u32) -> #crate_::metadata_ir::RuntimeApiMetadataIR
+			pub fn runtime_metadata #impl_generics (
+				impl_version: u32,
+				mut impl_deprecations: #crate_::scale_info::prelude::collections::BTreeMap<&str, #crate_::metadata_ir::ItemDeprecationInfoIR>
```

Comment:
```suggestion
				mut impl_deprecations: #crate_::__private::BTreeMap<&str, #crate_::metadata_ir::ItemDeprecationInfoIR>
```

And then re-export it there.